### PR TITLE
TRD bug fix for mac m1

### PR DESCRIPTION
**Pull request by bazinski**
This fixes a couple of things.
Handles the case of processHBF returning false (should never happen)
Correctly checks the bounds of the incoming rdh payload to ensure it fits in the allocated buffer for the half cru in question.

@chiarazampolli it appears to fix grid problem, I could not replicate grid problem with nightly and this executable and library running on top of the nightly last night.

@pzhristov Shout if this fixes your problems. I still running it through my series of timeframes i use for testing.

### CODE REVIEW

**Comment by bazinski** [comment]
This is currently rebuilding on linux(intel) and mac(m1), passed before, i assume still will. The problem of the singular tf i have with a config in it, remains, but I suspect corruption in the tf, comparing to dev in a 3rd build. Answers to that will come before the ci finishes with this.

**Comment by bazinski** [comment]
@martenole regarding your query, the actual fix for m1 is line 196(new) vs 160(old) of CruRawReader.cxx, together with the unit64_t move to uint32_t, together with correctly handling the return case of false from processHBFs.

File: Detectors/TRD/reconstruction/src/CruRawReader.cxx [hunk 1]
```diff
@@ -114,26 +114,45 @@ void CruRawReader::OutputHalfCruRawData()
...
+
 bool CruRawReader::processHBFs(int datasizealreadyread, bool verbose)
 {
   if (mHeaderVerbose) {
-    LOG(info) << "PROCESS HBF starting at " << std::hex << (void*)mDataPointer;
+    LOG(info) << "PROCESS HBF starting at " << std::hex << (void*)mDataPointer << " already read in : " << datasizealreadyread;
   }
   mDataRDH = reinterpret_cast<const o2::header::RDHAny*>(mDataPointer);
-  mOpenRDH = reinterpret_cast<o2::header::RDHAny*>((char*)mDataPointer);
+  mOpenRDH = reinterpret_cast<const o2::header::RDHAny*>((const char*)mDataPointer);
```
Comment (martenole) [inline Detectors/TRD/reconstruction/src/CruRawReader.cxx:139]: Ok, as I said I don't really understand most of this pointer casting, like here the `reinterpret_cast<>` with an intermediate cast to `char*`. And also the rest here is very hard to understand for me. But if you have tested this and it fixes an actual issue then it is fine with me.

**Review by martenole — APPROVED** [review]

File: Detectors/TRD/reconstruction/src/CruRawReader.cxx (hunk 1 above)
Comment (bazinski) [reply Detectors/TRD/reconstruction/src/CruRawReader.cxx:139]: ;-) that is actually not used, a legacy from testing early on and how I was intending to go through the rdhs comprising the relevant data. I ended up simply combining the payload parts for each rdh into a singular cru payload and working with that.
The change in this pr is actually accidental. New reader is way way more clear as to what is going on, and faster.

**Review by pzhristov — APPROVED** [review]
I confirm that these changes fix the issue on Mac M1. I propose to merge them.

**Pull request merged**